### PR TITLE
script: Encode field names and filenames correctly when encoding `<form>` data

### PR DESCRIPTION
**Pull request by yezhizhen**
[Spec](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#multipart/form-data-encoding-algorithm)

1. Escape field name for non-file field correctly. Escape both field name and filenames for field field correctly.
2. Fixed a corner case bug in step 1 implementation.

Testing: Many new passing in existing WPT. I was still following the FileAPI, but somehow the fix is about form data. The tests should probably be moved to `forms` folder. But that should probably be discussed in upstream.

Part of #10778

### CODE REVIEW

File: tests/wpt/meta/custom-elements/form-associated/ElementInternals-setFormValue.html.ini [hunk 1]
```diff
@@ -2,113 +2,53 @@
   [Newline normalization - \\n in name (urlencoded)]
     expected: FAIL
 
-  [Newline normalization - \\n in name (formdata)]
-    expected: FAIL
-
   [Newline normalization - \\r in name (urlencoded)]
```
Comment (yezhizhen) [inline tests/wpt/meta/custom-elements/form-associated/ElementInternals-setFormValue.html.ini:5]: I guess we have the same problem for urlencoded form. Checking this later.

File: components/script/dom/html/htmlformelement.rs [hunk 2]
```diff
@@ -2003,7 +2003,6 @@ pub(crate) fn encode_multipart_form_data(
                 // This character isn't LF but is
                 // preceded by CR
                 _ if prev == '\r' => {
-                    buf.push('\r');
```
Comment (yezhizhen) [inline components/script/dom/html/htmlformelement.rs:2006]: This was buggy before. For example, `\rA` would generate `\r\r\nA` previously. New passing.

File: components/script/dom/html/htmlformelement.rs [hunk 3]
```diff
@@ -2018,6 +2017,19 @@ pub(crate) fn encode_multipart_form_data(
...
+    // For field names and filenames for file fields, the result
+    // of the encoding must be escaped by replacing:
+    //   0x0A (LF) bytes with `%0A`,
+    //   0x0D (CR) bytes with `%0D`,
+    //   0x22 (") bytes with `%22`.
+    // The user agent must not perform any other escapes.
+    fn escape_for_header(s: &DOMString) -> String {
+        s.str()
+            .replace('\r', "%0D")
+            .replace('\n', "%0A")
```
Comment (TimvdLippe) [inline components/script/dom/html/htmlformelement.rs:2028]: Ultra-nit: the order is different here than what's written in the spec

**Review by TimvdLippe — APPROVED** [review]
